core/src/main/java/com/sequenceiq/cloudbreak/service/AbstractOrganizationAwareResourceService.java

```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    @Override
+    public T getByNameForOrganization(String name, Organization organization) {
+        T object = repository().findByNameAndOrganization(name, organization);
+        if (object == null) {
+            throw new NotFoundException(String.format("No %s found with name '%s'", resourceName(), name));
+        }
+        return object;
+    }
+
+    @Override
+    public T getByNameForOrganization(String name, Long organizationId) {
+        Organization organization = organizationService().get(organizationId);
+        return getByNameForOrganization(name, organization);
+    }
+
+    @Override
+    public T getByNameFromUsersDefaultOrganization(String name) {
+        Organization organization = organizationService().getDefaultOrganizationForCurrentUser();
+        return getByNameForOrganization(name, organization);
+    }
+
+    @Override
+    public Set<T> listByOrganization(Long organizationId) {
+        Organization organization = organizationService().get(organizationId);
+        return repository().findByOrganization(organization);
+    }
+
+    @Override
+    public Set<T> listByOrganization(Organization organization) {
+        return repository().findByOrganization(organization);
+    }
+
+    @Override
+    public Set<T> listForUsersDefaultOrganization() {
+        Organization organization = organizationService().getDefaultOrganizationForCurrentUser();
+        return repository().findByOrganization(organization);
+    }
+
+    @Override
+    public T delete(T resource) {
+        if (canDelete(resource)) {
+            repository().delete(resource);
+        }
+        return resource;
+    }
+
+    @Override
+    public T deleteByNameFromOrganization(String name, Long organizationId) {
+        T toBeDeleted = organizationId == null ? getByNameFromUsersDefaultOrganization(name) : getByNameForOrganization(name, organizationId);
+        return delete(toBeDeleted);
+    }
+
+    @Override
+    public T deleteByNameFromDefaultOrganization(String name) {
+        return deleteByNameFromOrganization(name, null);
+    }
+
+    private void setOrganization(T resource, IdentityUser identityUser, Long organizationId) {
+        User user = userService().getOrCreate(identityUser);
+        Organization organization;
+        if (organizationId != null) {
+            Set<Organization> usersOrganizations = organizationService().retrieveForUser(user);
+            organization = organizationService().get(organizationId);
+            if (!usersOrganizations.contains(organization)) {
+                throw new NotFoundException("Organization not found for user.");
+            }
+        } else {
+            organization = organizationService().getDefaultOrganizationForUser(user);
+        }
+        resource.setOrganization(organization);
+    }
+
+    protected abstract OrganizationResourceRepository<T, Long> repository();
+
+    protected abstract TransactionService transactionService();
```

Comment:
Can we remove these methods and inject TransactionService, OrganizationService, UserService into this abstract class with protected modifier?

Reply:
ok